app/controllers/user_auth_controller.rb

```diff
@@ -32,4 +32,18 @@ def user_not_authorized(exception)
   def authenticated_user_root_path
     current_user.only_invited? ? root_path : users_rdvs_path
   end
+
+  def verify_user_identity
+    return unless current_user.only_invited?
+    return if cookies.encrypted[:"user_#{current_user.id}_verified"] == true
+
+    session[:return_to_after_verification] = request.fullpath
+    redirect_to new_users_identity_verification_path
+  end
+
+  def set_user_identity_verified
+    cookies.encrypted[:"user_#{current_user.id}_verified"] = {
+      value: true, expires: 10.minutes.from_now
```

Comment:
À vérifier: est-ce que c’est contournable côté client? si oui, il faut le mettre dans le cookie chiffré.

Reply:
En fait j'ai réfléchi et Rails fait un digest du hash de ton cookie (valeur et expiration), donc c'est impossible de modifier la valeur ou l'expiration s'il est encrypté (s'il est signé aussi d'ailleurs vu que la signature changerait). Ce [post](https://dev.to/ayushn21/demystifying-cookie-security-in-rails-6-1j2f) explique bien le fonctionnement.